### ISSUE TITLE
Use 'vscode.open' instead of the 'open'

### ISSUE THREAD

**Issue by jdneo** [microsoft/vscode-azuretools#406]
Just a small suggestion, but using 'opn' is fine also...

For now, when we want to open a URL in the browser, we leverage the 'opn' module.

Actually, this module can be replaced with the VS Code embedded command, for example:

```typescript
export async function openUrl(url: string): Promise<void> {
    await vscode.commands.executeCommand("vscode.open", vscode.Uri.parse(url));
}
```

**Comment by jdneo** [comment]
@EricJizbaMSFT

**Comment by StephenWeatherford** [comment]
Huh...

**Comment by jdneo** [comment]
Guess this could be closed...

**Comment by bwateratmsft** [comment]
I'm going to reopen this; the use of `open` does not permit opening of URLs from in-browser Codespaces. See https://github.com/microsoft/vscode-docker/issues/2944 and https://github.com/microsoft/vscode-docker/issues/2943.

**Comment by bwateratmsft** [comment]
Modified the title slightly since we now use `open` instead of `opn`.

**Comment by bwateratmsft** [comment]
Related also to #867

**Comment by bwateratmsft** [comment]
I think that using `vscode.env.openExternal` may also get used as part of port forwarding with remote sessions.

**Comment by ejizba** [comment]
I don't think we'll fully switch to `openExternal` until that bug is fixed, but we could maybe special-case codespace scenarios

**Comment by bwateratmsft** [comment]
Is it frequent that we run into the URL parsing issues? If not, would it be practical to special case those and let the general case be `vscode.env.openExternal`?

**Comment by bwateratmsft** [comment]
This is fixed now by #1401
